scripts/core/usm.yml

```diff
@@ -255,9 +255,12 @@ params:
     - type: $x_device_handle_t
       name: hDevice
       desc: "[in] handle of the device object"
-    - type: $x_usm_desc_t*
-      name: pUSMDesc
-      desc: "[in][optional] USM memory allocation descriptor"
+    - type: const $x_usm_desc_t*
```

Comment:
If we have separate descriptors for device and host I think we should also refactor `ur_usm_mem_flags_t`. RIght now we have `INITIAL_PLACEMENT_DEVICE` and `INITIAL_PLACEMENT_HOST`. Probably it would be enough to have 
 a single `INITIAL_PLACEMENT` flag - and the placement would be determined whether it is set as part of device or host descriptor (similarly as in L0).
 
 Also, I think some of the flags do not make sense for host:
- `UR_USM_MEM_FLAG_DEVICE_READ_ONLY`
- `UR_USM_MEM_FLAG_WRITE_COMBINED` (L0 does not allow specifying WC for host)

Perhaps we should have a separate type for host/device descriptors?

Reply:
I like your suggestion. Perhaps this can be achieved by adding separate descriptors for host/device and linking them through the `pNext` parameter

Reply:
I added a test [here](https://github.com/oneapi-src/unified-runtime/pull/359/files#diff-7edcb8a2de0724f7f91e69ae9a7105611b9344ec3f9d19f90c22f0a12b1a25ebR40) To show how this might work.

Reply:
Yes, I think linking the descriptors is a good idea.